src/Google.Cloud.Functions.Invoker/HostingExtensions.cs

```diff
@@ -82,87 +78,128 @@ public static ILoggingBuilder AddFunctionsFrameworkConsoleLogging(this ILoggingB
-        /// The application configurers are then executed by <see cref="ConfigureApplicationForFunctionsFramework(IWebHostBuilder)"/>.
+        /// Adds the specified application configurers services.
+        /// The application configurers are then executed by <see cref="UseFunctionsFramework(IApplicationBuilder, WebHostBuilderContext)"/>.
         /// </summary>
-        /// <param name="builder">The web host builder to configure.</param>
+        /// <param name="services">The service collection to add services to.</param>
         /// <param name="configurers">The application configurers.</param>
-        /// <returns>The original builder, for method chaining.</returns>
-        public static IWebHostBuilder ConfigureFunctionsApplicationConfigurers(this IWebHostBuilder builder, IEnumerable<FunctionsApplicationConfigurer> configurers) =>
-            builder.ConfigureServices((context, services) => HostingInternals.ConfigureApplicationConfigurers(context, services, configurers));
+        /// <returns>The original service collection, for method chaining.</returns>
+        public static IServiceCollection AddFunctionsApplicationConfigurers(this IServiceCollection services, WebHostBuilderContext context, IEnumerable<FunctionsApplicationConfigurer> configurers)
+        {
+            HostingInternals.ConfigureApplicationConfigurers(context, services, configurers);
+            return services;
+        }
 
         /// <summary>
-        /// Configures the given web host builder to use the Functions Framework.
+        /// Configures the given app builder to use the Functions Framework.
         /// This method executes any application configurers before adding handlers for fixed paths to
         /// return "not found" responses for fixed paths (e.g. "favicon.ico") and setting the terminal
         /// handler to execute the target function.
         /// </summary>
-        /// <param name="builder">The web host builder to configure.</param>
+        /// <param name="app">The application builder to configure.</param>
         /// <returns>The original builder, for method chaining.</returns>
-        public static IWebHostBuilder ConfigureApplicationForFunctionsFramework(this IWebHostBuilder builder) =>
-            builder.Configure(HostingInternals.ConfigureApplication);
+        public static IApplicationBuilder UseFunctionsFramework(this IApplicationBuilder app, WebHostBuilderContext context)
+        {
+            HostingInternals.ConfigureApplication(context, app);
+            return app;
+        }
+
+        /// <summary>
+        /// Helper to configure Functions Framework with default settings.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="functionAssembly"></param>
+        /// <param name="args"></param>
+        /// <returns></returns>
+        public static IWebHostBuilder UseFunctionsFramework(this IWebHostBuilder builder, Assembly functionAssembly, string[] args)
```

Comment:
I originally considered having a method like this, but then thought that the only time you want to configure things manually is when you want some of this not to occur - in which case you wouldn't want to call the method.

The use cases for users calling any of these methods themselves are primarily if they're customizing the pipeline from scratch.

The other use cases is in FunctionTestServer Builder:)